### PR TITLE
Updated clock synchronization blog

### PR DESCRIPTION
**Pull request by prasadtiru**
I've now updated the blog with a new section where one does not have to rely on RTC when calibrating.

### CODE REVIEW

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md [hunk 1]
```diff
@@ -132,4 +134,44 @@ phy.clockCalib = mB/mA
...
+
+for i in range(noOfMeasurements):
+	phytimeA.append(phyA.time)
+	phytimeB.append(phyB.time)
+	time.sleep(interval)
+
+# compute slope based on measurements
+m = (phytimeB[noOfMeasurments-1] - phytimeB[0])/(phytimeA[noOfMeasurments-1] - phytimeA[0]) // slope
+```
+Now based on the computed slope (m), the modem A or modem B's `clockCalib` parameter can be set accordingly to match the other modem's clock as explained in previous section.
```
Comment (mchitre) [inline _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md:174]: You probably want to give the exact expression of what value you'd set on which modem.

**Review by mchitre — CHANGES_REQUESTED** [review]

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md [hunk 2]
```diff
@@ -10,14 +10,14 @@ thumbnail: "assets/img/stdma/stdma-unet.jpg"
 tags: [howto, time synchronization, clock drift, accurate clocks, clock offset]
 ---
 
-All UnetStack enabled modems have a “real-time clock” to keep track of time using a precise quartz crystal. Based on the defined frequency of this quartz, software running on the modem can precisely monitor the internal physical time given by parameter `phy.time` in the modem. 
+All UnetStack enabled modems have a Real-time Clock (RTC) to keep track of time using a precise quartz crystal. Based on the defined frequency of this quartz, software running on the modem can precisely monitor the internal physical time given by parameter `phy.time` in the modem.
```
[the remark this answers is not in the document]
Comment (notthetup) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md:13]: `phy.time` is the PHY clock
`phy.rtc` is the modem system (Linux) clock

`phy.rtc` gets corrected by NTP. `phy.time` is a sample-accurate clock for PHY only.

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md (hunk 1 above)
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: Fixed.

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md [hunk 3]
```diff
@@ -132,4 +134,44 @@ phy.clockCalib = mB/mA
 
 ```
 
-Note the parameters used in the above code snippet such as `interval` and `noOfMeasurements` can be varied as per the user's need. The code snippet shown in this blog is to serve as a reference only.
+# Calibrating clock without relying on RTC
+
+It is possible that the RTC is no more accurate than the physical clock. In that case, it is obvious that it should not be used as a reference time source. Instead, we can measure and compare the clocks of the two modems directly. In this section, we describe through a simple python script on how one can achieve this.
```
[the remark this answers is not in the document]
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: Added appropriate explanation for this now.

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md [hunk 4]
```diff
@@ -132,4 +134,44 @@ phy.clockCalib = mB/mA
 
 ```
 
-Note the parameters used in the above code snippet such as `interval` and `noOfMeasurements` can be varied as per the user's need. The code snippet shown in this blog is to serve as a reference only.
+# Calibrating clock without relying on RTC
```
[the remark this answers is not in the document]
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md:139]: Added now.

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: Updated the description now. My understanding is following: 

The RTC is provided by a crytsal and the `phy.time` is just a microsecond counter in software for local counting needs. 

I have updated the description based on this understanding. Please correct me if I am wrong.

[the remark this answers is not in the document]
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: The hardware source of `phy.rtc` is a crystal that is being used by Linux to count time (i.e., this is probably on a TX2 board). The hardware source for `phy.time` is a different crystal (e.g., an OCXO) which is probably on Pillio. Am I correct ?

[the remark this answers is not in the document]
Comment (mchitre) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: Correct.

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md [hunk 5]
```diff
@@ -10,14 +10,14 @@ thumbnail: "assets/img/stdma/stdma-unet.jpg"
 tags: [howto, time synchronization, clock drift, accurate clocks, clock offset]
 ---
 
-All UnetStack enabled modems have a “real-time clock” to keep track of time using a precise quartz crystal. Based on the defined frequency of this quartz, software running on the modem can precisely monitor the internal physical time given by parameter `phy.time` in the modem. 
+All UnetStack enabled modems have a Real-time Clock (RTC) to keep track of modem's system time using a precise quartz crystal. A separate microsecond counter is provided by `phy.time` for `Physical` agent in UnetStack.
```
Comment (mchitre) [inline _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md:13]: This one still doesn't work for me. Why is important that RTC is precise or not?  It's `phy.time` that this article is about!

**Review by mchitre — CHANGES_REQUESTED** [review]

File: _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md (hunk 5 above)
Comment (prasadtiru) [reply _posts/2021-06-28-Synchronizing-physical-clocks-among-different-UnetStack-based-modems.md]: Modified the introduction now to be more clear now.

**Review by notthetup — APPROVED** [review]

**Review by mchitre — APPROVED** [review]

**Pull request merged**